src/routes/Home.jsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { BsSearch } from 'react-icons/bs';
+import { getCurrencies, getCurrencyPrice } from '../redux/currency/currencySlice';
+import CurrencyComponent from '../components/Currency';
+import { changeMoney } from '../redux/money/moneySlice';
+
+const Home = () => {
+  const dispatch = useDispatch();
+  const currencies = useSelector((state) => state.currency.allCurrencies);
+  const price = useSelector((state) => state.currency.usdPrice);
+  const money = useSelector((state) => state.money.value);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    dispatch(getCurrencies());
+    dispatch(getCurrencyPrice());
+  }, [dispatch]);
+
+  const handleSearch = () => {
+    const search = document.querySelector('.search');
+    const error = document.querySelector('.error');
+    if (search.value === '') {
+      error.textContent = 'Please enter a currency code';
+      setTimeout(() => {
+        error.textContent = '';
+      }, 3000);
+    } else {
+      error.textContent = '';
+      if (!currencies[search.value]) {
+        error.textContent = 'Currency not found';
+        setTimeout(() => {
+          error.textContent = '';
+        }, 3000);
+      } else {
+        error.textContent = '';
+        navigate(`/${search.value}`);
+      }
+    }
+  };
+
+  const handleMoney = (e) => {
+    dispatch(changeMoney(e.target.value));
+  };
+
+  let color = 0;
+
+  return (
+    <div className="App">
+      <div className="search-bar">
+        <input type="text" className="search" placeholder="search by code" />
+        <button type="button" className="search-btn" onClick={handleSearch}>
+          <BsSearch />
+        </button>
```

Comment:
- [ ] Good job adding a feature that can filter data retrieved from API. However I noticed an issue when trying to search some cryptocurrencies in the list of categories. From your UI currency's code is in **CAPITAL** letter but when I type anyone from the list in **CAPITAL** it is the `Error_message` which displayed instead getting the typed item. 
I suggest you improve this feature for example `to get data even if it is typed in Capital`, `specify in your placeholder text` or `specify in your error message text`,...